### ISSUE TITLE
fastify.auth(functions: fastifyAuth.AuthFunction[]) is missing second parameter in type declaration

### ISSUE THREAD

**Issue by espenmeidell** [fastify/fastify-auth#59]
## 🐛 Bug Report

The Typescript type for [`fastify.auth`](https://github.com/fastify/fastify-auth/blob/87947b90e0f21deab6cc027c570ca66693ee1888/fastify-auth.d.ts#L34) is missing the optional second parameter that allows us to change the relationship between the authentication functions from `or` to `and` (as described in the [readme](https://github.com/fastify/fastify-auth/blob/master/README.md)).

## To Reproduce

Use fastify-auth with Typescript. Attempt to change the relation between two auth functions.

```ts
fastify.get(
    '/auth-test',
    {
      schema: authSchema,
      preHandler: fastify.auth([fastify.jwtTokenCheck, fastify.userIsAdmin], {relation: 'and'}),
    },
    async () => {
      return { status: 'ok' };
    },
  );
```

## Expected behavior

We would expect this to work as in the readme, but Typescript expects only one parameter for the function.

Paste the results here:

```
error TS2554: Expected 1 arguments, but got 2.
```
## Your Environment

- *node version*: 10
- *fastify version*:^2.10.0
- *fastify-auth version*: ^0.6.0
- *os*: Mac

**Comment by mcollina** [comment]
Would you like to send us a PR?
Il 14 nov 2019, 13:53 +0000, Espen Meidell <notifications@github.com>, ha scritto:

**Comment by espenmeidell** [comment]
> Would you like to send us a PR? 

https://github.com/fastify/fastify-auth/pull/60 solves the issue 😊

**Comment by espenmeidell** [comment]
Solved in fastify-auth@0.6.1
